src/components/container/RelativeActivity/RelativeActivity.tsx

```diff
@@ -0,0 +1,111 @@
+import { faChevronUp } from "@fortawesome/free-solid-svg-icons";
+import { startOfDay } from "date-fns";
+import React, { useContext } from "react";
+import { useState } from "react";
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+import getDayKey from "../../../helpers/get-day-key";
+import { getDailyDataTypeDefinition } from "../../../helpers/query-daily-data";
+import { ActivityMeter, CardTitle, DateRangeContext, } from "../../presentational";
+import "./RelativeActivity.css"
+import language from "../../../helpers/language";
+import { useInitializeView } from "../../../helpers/Initialization";
+import { RelativeActivityContext } from "../RelativeActivityDayCoordinator/RelativeActivityDayCoordinator";
+import { RelativeActivityDataType, RelativeActivityQueryResult } from "../../../helpers";
+import { queryRelativeActivity } from "../../../helpers/relative-activity";
+
+export interface RelativeActivityProps {
+    dataTypes?: RelativeActivityDataType[];
+    useContext?: boolean;
+    previewState?: "Default";
+    title?: string;
+    innerRef?: React.Ref<HTMLDivElement>
+    date?: Date;
+}
+
+export default function (props: RelativeActivityProps) {
+    let [results, setResults] = useState<{ [key: string]: RelativeActivityQueryResult } | undefined>(undefined);
+    let relativeActivityContext = useContext(RelativeActivityContext);
+
+    let dataTypes: RelativeActivityDataType[] = props.dataTypes || [];
+    if (props.useContext && relativeActivityContext?.dataTypes) {
+        dataTypes = relativeActivityContext.dataTypes;
+    }
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let date = props.date ?? dateRangeContext?.intervalStart ?? startOfDay(new Date());
+
+    function loadData() {
+        function transformResults(results: { [key: string]: { [key: string]: RelativeActivityQueryResult } } | undefined) {
+            if (!results) return;
+            let computedResult: { [key: string]: RelativeActivityQueryResult } = {};
+            dataTypes.forEach(dataType => {
+                computedResult[dataType.dailyDataType] = results[dataType.dailyDataType][getDayKey(date!)];
+            });
+            return computedResult;
+        }
+
+        setResults(undefined);
+        if (props.useContext) {
+            setResults(transformResults(relativeActivityContext!.data));
+            return;
+        }
+        queryRelativeActivity(date!, date!, dataTypes, !!props.previewState).then(results => {
+            setResults(transformResults(results));
+        });
+    }
+
+    useInitializeView(() => {
+        loadData();
+    }, ["externalAccountSyncComplete"], [dateRangeContext, relativeActivityContext, props.date, props.dataTypes], 0);
+
+    if (results == null || !Object.keys(results).length) {
```

Comment:
Minor: I think this may work because of the nature of `==`, but `result` is only ever `undefined` or an object, so it may be more appropriate to just use `!result` here.